Rakefile

```diff
@@ -130,22 +130,30 @@ REGEN_EXCLUDES = ['azure_sdk']
 desc 'Azure Resource Manager related tasks which often traverse each of the arm gems'
 namespace :arm do
   desc 'Delete ./pkg for each of the Azure Resource Manager projects'
-  task :clean do
-    each_gem do
+  task :clean, [:key] do |_, args|
+    each_gem do |dir|
+      if !should_i_continue(args[:key], dir)
```

Comment:
should we be centralizing this logic more, like applying filtering at "each_child" method level?
 https://github.com/sarangan12/azure-sdk-for-ruby/blob/021d8322512ac3fe0a835cbb01d025d68e298bde/Rakefile#L450 and have a file that allows us to control which gems to generate/select? 
Having a parameter like it's added in this PR, is ok for individual gem generation and it's good, but I'm thinking if we should think a bit broader enabling generation of multiple gems too, so it facilitates our own steps. We could have a file similar to the "GEMS_to_release" which has a checked in sample with all the gem list, but with the purpose of editing locally for generation. Also, regarding "GEMS_to_release" file, I think we should enable our tasks to consume it, so that when we indicate a few gems to get released, the jobs only generate and build those gems (instead of doing everything and then skip the release step).

Reply:
I think adding GEMS_TO_RELEASE to this would be overkill. It could be easily achieved by using the && command. Let us see if we really need that feature and then think about adding it.

Reply:
Ok, as per my comment above I wouldn't tie it with GEMS_TO_RELEASE only, that comment was more of a convenience thing, as I think it can save some time (not building all gems), so it'd be good to consider. 
Regarding not modifying the approach right now, I'll leave it up to you, design wise I prefer the idea of centralizing logic and selecting from the start, instead of having code that skips gems repeated in multiple tasks.